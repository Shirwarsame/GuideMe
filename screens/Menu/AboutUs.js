import React from "react";
import { View, Text, StyleSheet, SafeAreaView, SectionList } from "react-native";
import { ScrollView } from "react-native-gesture-handler";
import { Paragraph } from 'react-native-paper';
import { Icon, Button } from "native-base";
import { Feather } from "@expo/vector-icons";

export const renderSeparator = () => {
    return (
        <View
            style={{
                height: 1,
                width: "100%",
                backgroundColor: "#353A50",
            }}
        />
    );
};

function AboutUs(props) {

    const goToMenu = () => {
        props.navigation.openDrawer();
    };

    const data = [{
        title: "Team - Github",
        data: ["Mohanad Arafe - mohanadarafe",
            "Christophe Shaka Bahenduzi - csbduzi",
            "Kenza Boulisfane - bkenza",
            "Jérémie Chouteau - JayChoteau",
            "Baddredine Loulidi - badred123",
            "Firas Sawan - firassawan",
            "Alain Job Uzarama Uwe - alainjobs",
            "Shirwa Warsame - shirwarsame",
            "Hussain Witwit - hussainwitwit"
        ]
    }]

    return (
        <View style={styles.container}>
            <View style={styles.menuButtonContainer}>
                <Button transparent style={styles.menuButton} onPress={goToMenu}>
                    <Feather name="menu" style={styles.icon} />
                </Button>
            </View>

            <Text style={styles.mainLabel}>About Us</Text>
            <Text style={styles.aboutLabel}>Our App</Text>

            <View style={styles.scrollViewTextContainer}>
            <ScrollView style={{flex:1}}>
                 <Paragraph style={styles.aboutText}>GuideMe is a software application made for commuters of Concordia University. The purpose of the application is to serve users a map for both the Loyola & Sir George Williams Campuses. The application gives a new refined way to navigate on campus offering route recommendations & various methods of travelling.
                  </Paragraph>
                 </ScrollView>
            </View>
           
            <Text style={styles.teamLabel}>Our Team</Text>

            <SafeAreaView style={styles.scrollTextContainer}>
                <SectionList
                    sections={data}
                    renderItem={({ item }) => <Text style={styles.listItem}>{item}</Text>}
                    renderSectionHeader={({ section }) => <Text style={styles.sectionHeader}>{section.title}</Text>}
                    keyExtractor={(item, index) => index}
                    ItemSeparatorComponent={renderSeparator}
                />
            </SafeAreaView>


        </View >
    );
}

export const styles = StyleSheet.create({
    container: {
        alignItems: "center",
        justifyContent: "space-between",
        height: "100%",
        width: "100%",
        backgroundColor: "#2A2E43",
        flex:1
    },
    scrollViewTextContainer:{
        bottom: "57%",
        height: 150,
        width: 415,
    },
    mainLabel: {
        color: "#FFFFFF",
        position: "absolute",
        fontSize: 25,
        fontWeight: "bold",
        fontFamily: "encodeSansExpanded",
        top: "15%"
    },
    aboutLabel: {
        color: "#FFFFFF",
        position: "absolute",
        fontSize: 18,
        fontWeight: "bold",
        fontFamily: "encodeSansExpanded",
        top: "22%"
    },
    aboutText: {
        position: "absolute",
        alignSelf: "center",
        fontSize: 12,
        fontFamily: "encodeSansExpanded",
        color: "#D3D3D3",
        top: "27%",
        width: "85%",
        textAlign: "center"
    },
    teamLabel: {
        color: "#FFFFFF",
        position: "absolute",
        fontSize: 18,
        fontFamily: "encodeSansExpanded",
        top:"43%"
    },
    icon: {
        height: "100%",
        width: "100%",
        flexDirection: "row",
        left: "6%",
        color: "#FFFFFF",
        fontSize: 35
    },
    menuButton: {
        height: "100%",
        width: "100%",
        flexDirection: "row",
    },
    menuButtonContainer: {
        width: "100%",
        height: "6%",
        flexDirection: "column",
        justifyContent: "space-around",
        alignContent: "center",
        alignItems: "center",
        top: "7%"
    },

    scrollTextContainer: {
        width: "100%",
        height: "52%",
        bottom: "0%",
        position: "absolute",
    },
    sectionHeader: {
        paddingTop: 2,
        paddingLeft: 22,
        paddingRight: 10,
        paddingBottom: 2,
        fontSize: 18,
        fontWeight: "bold",
        color: "#ffffff",
        backgroundColor: "#353A50",
        fontFamily: "encodeSansExpanded"
    },
    listItem: {
        padding: 10,
        fontSize: 12,
        height: 44,
        paddingLeft: 22,
        fontFamily: "encodeSansExpanded",
        color: "#D3D3D3",
    },

});

export default AboutUs;
