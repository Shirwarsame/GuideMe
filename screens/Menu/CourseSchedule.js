import { Feather } from "@expo/vector-icons";
import PropTypes from "prop-types";
import React, { useEffect } from "react";
import { AsyncStorage, FlatList, StyleSheet, Text, TouchableOpacity, View } from "react-native";
import { CheckBox, ListItem } from "react-native-elements";
import { ScrollView } from "react-native-gesture-handler";

/**
 * US22 - As a user, I would like to connect my direction to next class to my google calendar. #31
 * US23 - As a user, when connecting the direction to the next class in my google calendar, I would like to map those directions to multiple calendars. #32
 * US24 - As a user, I would like to find the location of a classroom from the next calendar event.
 * 
 * Description: The CourseSchedule method will 
 * render the screen for the CourseSchedule screen, 
 * which is composed of a scroll list that will hold
 * the information of the courses retrived from the Google Calendar API.
 * The items from the sectionList will be pressable for the uesr 
 * to navigate to the detail screen and get the directions to the class.
 */

/**Prop passed
 * @param  {} navigation props.navigation is the name of the object from Navigator library
 */
function CourseSchedule(props) {
    const [accessToken, setAccessToken] = React.useState(null);
    const [selectedCalendarId, setSelectedCalendarId] = React.useState(null);
    const [calendarEventsList, setCalendarEventsList] = React.useState(null);
    const [switchVal, setSwitchVal] = React.useState(false);

    const getCalendarId = async () => {
        let calendarId = await AsyncStorage.getItem("calendarId");
        setSelectedCalendarId(calendarId);
    };

    const getAccessToken = async () => {
        let accessToken = await AsyncStorage.getItem("accessToken");
        setAccessToken(accessToken);
    };

    const getSwitchValue = async () => {
        let switchValue = await AsyncStorage.getItem('switchVal');
        setSwitchVal(switchValue);
    };

    const getCalendarEvents = async () => {
        let calendarEvents = await fetch('https://www.googleapis.com/calendar/v3/calendars/' + selectedCalendarId + '/events', {
            headers: { Authorization: `Bearer ${accessToken}` },
        });
        let resp = await calendarEvents.json();
        setCalendarEventsList(getFilteredGoogleCalendarEvents(resp));
    }

    const getFilteredGoogleCalendarEvents = (resp) => {
        var filteredList = resp.items.map(element => {
            return { id: element.id, summary: element.summary, description: element.description, location: element.location };
        });
        return filteredList;
    };

    const CourseScheduleScreen = true;

    /**
     * The method will slide the side menu from the right side of the screen
     * @param  {} =>{props.navigation.openDrawer(
    */
    const goToMenu = () => {
        props.navigation.openDrawer();
    };

    const goToDoubleSearch = (item) => {
        props.navigation.navigate("DoubleSearch", { CourseScheduleScreen: CourseScheduleScreen, CourseScheduleLocation: item.location });
    }

    useEffect(() => {
        const intervalId = setInterval(() => {
            getCalendarId();
            getAccessToken();
            getSwitchValue();
        }, 1);
        getCalendarEvents();
        return () => clearInterval(intervalId);
    }, [selectedCalendarId, accessToken]);

    return (
        <View style={styles.container}>
            <View style={styles.menuButtonContainer}>
                <TouchableOpacity style={styles.menuButton} onPress={goToMenu}>
                    <Feather name="menu" style={styles.icon} />
                </TouchableOpacity>
            </View>
            <Text style={styles.mainLabel}>My Course Schedule</Text>
            <ScrollView style={styles.scrollTextContainer}>
                {(switchVal == "true" && calendarEventsList) &&
                    <FlatList
                        data={calendarEventsList}
                        keyExtractor={(item) => item.id}
                        renderItem={({ item }) => (
                            <ListItem
                                title={item.summary}
                                titleStyle={{ color: "white", paddingLeft: 20 }}
                                subtitle={item.location !== undefined ? item.description : "Please enter a location in your Google Calendar"}
                                rightTitleStyle={{ color: "white", paddingRight: 20 }}
                                containerStyle={{ backgroundColor: "#2A2E43" }}
                                subtitleStyle={{ color: "grey", paddingLeft: 20 }}
                                rightIcon={
                                    <CheckBox
                                        size={30}
                                        iconRight
                                        iconType='material'
                                        uncheckedIcon='arrow-forward'
                                        uncheckedColor="#3ACCE1"
                                        onPress={() => { goToDoubleSearch(item) }}
                                    />}
                            />
                        )}
                    />
                }
            </ScrollView>
        </View >
    );
}


CourseSchedule.propTypes = {
    navigation: PropTypes.object,
    openDrawer: PropTypes.func,
    navigate: PropTypes.func
};

export const styles = StyleSheet.create({
    container: {
        alignItems: "center",
        justifyContent: "space-between",
        height: "100%",
        width: "100%",
        backgroundColor: "#2A2E43"
    },
    mainLabel: {
        color: "#FFFFFF",
        position: "absolute",
        fontSize: 25,
        fontWeight: "bold",
        fontFamily: "encodeSansExpanded",
        top: "15%"
    },
    icon: {
        alignSelf: "center",
        color: "#FFFFFF",
        fontSize: 35,
    },
    menuButton: {
        height: "100%",
        width: "20%",
        flexDirection: "row",
        justifyContent: "center"
    },
    menuButtonContainer: {
        width: "100%",
        height: "6%",
        top: "7%",
    },
    scrollTextContainer: {
        width: "100%",
        height: "75%",
        bottom: "0%",
        position: "absolute",
    },
    sectionHeader: {
        paddingTop: 2,
        paddingLeft: 22,
        paddingRight: 10,
        paddingBottom: 2,
        fontSize: 18,
        fontWeight: "bold",
        color: "#ffffff",
        backgroundColor: "#353A50",
        fontFamily: "encodeSansExpanded"
    },
    listItem: {
        padding: 10,
        fontSize: 12,
        height: 65,
        paddingLeft: 22,
        fontFamily: "encodeSansExpanded",
        color: "#D3D3D3",
    },
    line: {
        height: 1,
        width: "100%",
        backgroundColor: "#353A50",
    },

});

export default CourseSchedule;
