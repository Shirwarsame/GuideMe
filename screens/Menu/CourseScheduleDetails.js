import React from "react";
import { View, Text, StyleSheet } from "react-native";
import { Icon, Button } from "native-base";
import PropTypes from "prop-types";


function CourseScheduleDetails (props) {

    const goBack = () => {
        props.navigation.goBack();
    };

    // const goToDoubleSearch = () => {
    //     props.navigation.navigate("DoubleSearch");
    // }

    return (
        <View style={styles.container}>
            <View style={styles.menuButtonContainer}>
                <Button transparent style={styles.menuButton} onPress={goBack}>
                    <Icon name="md-arrow-round-back" style={styles.icon} />
                </Button>
            </View>

            <Text style={styles.mainLabel}>Class Details</Text>

            <Button transparent style={styles.routeButton}><Text style={styles.viewRouteText}>View Route</Text></Button>
        </View >
    );
}

CourseScheduleDetails.propTypes = {
    navigation: PropTypes.object,
    goBack: PropTypes.func
};

export const styles = StyleSheet.create({
    container: {
        alignItems: "center",
        justifyContent: "space-between",
        height: "100%",
        width: "100%",
        backgroundColor: "#2A2E43"
    },
    mainLabel: {
        color: "#FFFFFF",
        position: "absolute",
        fontSize: 25,
        fontWeight: "bold",
        fontFamily: "encodeSansExpanded",
        top: "15%"
    },
    icon: {
        height: "100%",
        width: "100%",
        flexDirection: "row",
        left: "6%",
        color: "#FFFFFF",
        fontSize: 35
    },
    menuButton: {
        height: "100%",
        width: "100%",
        flexDirection: "row",
    },
    menuButtonContainer: {
        width: "100%",
        height: "6%",
        flexDirection: "column",
        justifyContent: "space-around",
        alignContent: "center",
        alignItems: "center",
        top: "7%"
    },
    routeButton: {
        width: "90%",
        height: "8%",
        fontSize: 25,
        bottom: "8%",
        justifyContent: "center",
        backgroundColor: "#3ACCE1",
        borderRadius: 10,
    },
    viewRouteText: {
        color: "white",
        fontSize: 14
    }
});

export default CourseScheduleDetails;
