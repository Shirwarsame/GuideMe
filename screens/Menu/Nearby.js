import React, { useState } from "react";
import { View, Text, StyleSheet, AsyncStorage, FlatList, Dimensions } from "react-native";
import { Icon, Button } from "native-base";
import { Feather } from "@expo/vector-icons";



function Nearby(props) {

    const [fromBottomMenu, setFromBottoMenu] = React.useState({fromBottomMenu: bottomMenu?bottomMenu:null});

    const goBack = () => {
        setFromBottoMenu(null);
        console.log(bottomMenu);
        props.navigation.goBack();
    };

    const goToMenu = () => {
        props.navigation.openDrawer();
    };

    const bottomMenu = props.navigation.getParam("bottomMenu", "null");
    console.log(bottomMenu);

    // const data = [{ key: "A" }, { key: "B" }, { key: "C" }];
    // const numColumns = 2;


    // const renderItem = ({ item, index }) => {

    //     return (
    //         <View
    //             style={{
    //                 flex: 1,
    //                 margin: 5,
    //                 minWidth: 170,
    //                 maxWidth: 223,
    //                 height: 200,
    //                 maxHeight: 200,
    //                 backgroundColor: '#353A50',
    //             }}
    //         />
    //     );
    // }


    return (
        <View style={styles.container}>
            {bottomMenu
                ? <Icon name="ios-arrow-down" style={styles.arrowDown} onPress={goBack} />
                : <Button transparent style={styles.menuButton} onPress={goToMenu}>
                    <Feather name="menu" style={styles.icon} />
                </Button>
            }
            <Text style={styles.mainLabel}>Points of Interest</Text>

            {/* <FlatList
                onEndReachedThreshold={0}

                contentContainerStyle={styles.list}
                data={[
                    { key: 'a' },
                    { key: 'b' },
                    { key: 'c' },
                    { key: 'd' },
                    { key: 'e' },
                    { key: 'f' },
                    { key: 'g' },
                    { key: 'h' },
                    { key: 'i' },
                    { key: 'j' },
                ]}
                renderItem={renderItem}
            /> */}

        </View >
    );
}

export const styles = StyleSheet.create({
    container: {
        alignItems: "center",
        justifyContent: "space-between",
        height: "100%",
        width: "100%",
        backgroundColor: "#2A2E43"
    },
    mainLabel: {
        color: "#FFFFFF",
        position: "absolute",
        fontSize: 25,
        fontWeight: "bold",
        fontFamily: "encodeSansExpanded",
        top: "15%"
    },
    arrowDown: {
        color: "#ffffff",
        top: "5%",
        fontSize: 54,
        position: "absolute"
    },

    list: {
        justifyContent: 'center',
        flexDirection: 'row',
        flexWrap: 'wrap',
        top: "50%"
    },

    icon: {
        height: "100%",
        width: "100%",
        flexDirection: "row",
        left: "6%",
        color: "#FFFFFF",
        fontSize: 35

},
menuButton: {
    height: "100%",
    width: "100%",
    flexDirection: "row",
},
menuButtonContainer: {
    width: "100%",
    height: "6%",
    flexDirection: "column",
    justifyContent: "space-around",
    alignContent: "center",
    alignItems: "center",
    top: "7%"
},
});

export default Nearby;
