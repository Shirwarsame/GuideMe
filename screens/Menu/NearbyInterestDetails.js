import React from "react";
import { View, Text, StyleSheet, Image } from "react-native";
import { Icon, Button } from "native-base";


/**Prop passed
 * @param  {} navigation props.navigation is the name of the object from Navigator library
 */

function NearbyInterestDetails(props) {

    const goBack = () => {
        props.navigation.goBack();
    }

    const NearbyInterestDetailsScreen = true;
    const goToDoubleSearch = () => {
        props.navigation.navigate("DoubleSearch", { NearbyInterestDetailsScreen: NearbyInterestDetailsScreen });
    }

    const name = props.navigation.getParam("name", null);
    const rating = props.navigation.getParam("rating", null);
    const photoref = props.navigation.getParam("photoref", null);

    return (
        <View style={styles.container}>
           

            <View style={styles.imageContainer}>
                <Image style={styles.placeImage} source={{ uri: photoref }} />
            </View>
            <Text style={styles.mainLabel}>{name ? name : "N/A"}</Text>
            <Text style={styles.reviewLabel}>{rating ? rating : "N/A"}</Text>

            <View style={styles.backArrowContainer}>
                <Button transparent style={styles.backArrow} onPress={goBack}>
                    <Icon name="md-arrow-round-back" style={styles.icon}></Icon>
                </Button>
            </View>

            <Button transparent style={styles.routeButton} onPress={goToDoubleSearch}><Text style={styles.viewRouteText}>View Route</Text></Button>
        </View >
    );
}

export const styles = StyleSheet.create({
    container: {
        alignItems: "center",
        justifyContent: "space-between",
        height: "100%",
        width: "100%",
        backgroundColor: "#2A2E43"
    },
    mainLabel: {
        color: "#FFF",
        fontSize: 23,
        fontFamily: "encodeSansExpanded",
        paddingBottom: 10,
        top: "14%",
        bottom: "2%",
        fontWeight: "bold",
        justifyContent: "center",
        position: "absolute"
    },
    icon: {
        position: "absolute",
        color: "#FFFFFF",
        alignSelf: "center",
        fontSize: 35
    },
    routeButton: {
        width: "90%",
        height: "8%",
        fontSize: 25,
        bottom: "8%",
        justifyContent: "center",
        backgroundColor: "#3ACCE1",
        borderRadius: 10,
    },
    viewRouteText: {
        color: "white",
        fontSize: 14,
    },
    backArrow: {
        height: "100%",
        width: "100%",
        flexDirection: "row",
        left: "10%"
    },
    backArrowContainer: {
        width: "100%",
        height: "6%",
        flexDirection: "column",
        justifyContent: "space-around",
        alignContent: "center",
        alignItems: "center",
        top: "7%"
    },

    imageContainer: {
        width: "100%",
        height: "32%",
        top: "0%",
        position: "absolute",
        opacity: 0.3
    },

    placeImage: {
        width: "100%",
        height: "100%",
        top: "0%",
        position: "relative"
    },
    mainLabel: {
        color: "#FFFFFF",
        left: "5%",
        position: "absolute",
        fontSize: 30,
        fontWeight: "bold",
        fontFamily: "encodeSansExpanded",
        top: "19%"
    },
    subLabel: {
        position: "absolute",
        top: "24%",
        left: "5%",
        color: "#FFFFFF",
        fontSize: 11,
        fontWeight: "bold",
        fontFamily: "encodeSansExpanded"
    },
    reviewLabel: {
        position: "absolute",
        top: "27%",
        left: "5%",
        color: "#FFFFFF",
        fontSize: 20,
        fontFamily: "encodeSansExpanded"
    },

});

export default NearbyInterestDetails;
