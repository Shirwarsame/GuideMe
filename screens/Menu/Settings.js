import { Feather } from "@expo/vector-icons";
import * as Google from 'expo-google-app-auth';
import PropTypes from "prop-types";
import React, { useEffect } from "react";
import { AsyncStorage, FlatList, StyleSheet, Switch, Text, TouchableOpacity, View } from "react-native";
import { CheckBox, ListItem } from "react-native-elements";
import { ScrollView } from "react-native-gesture-handler";
import { sideMenuStyle } from "../../assets/styling/sideMenuStyling";

/**
 * Description: This method holds the toggle switches 
 * that will compose the setting screen. The screen will have 
 * also have accessibility to a Google Account Login screen.
 */

/**Prop passed
* @param  {} navigation props.navigation is the name of the object from Navigator library
*/
function Settings(props) {
    const [switchVal1, setSwitchVal1] = React.useState(false);
    const [switchVal2, setSwitchVal2] = React.useState(false);
    const [calendarList, setCalendarList] = React.useState(null);
    const [isConnected, setIsConnected] = React.useState({ checked: [] });
    const [accessToken, setAccessToken] = React.useState("");

    var switchLabel1 = switchVal1 ? "ON" : "OFF";
    var switchLabel2 = switchVal2 == "true" ? "ON" : "OFF";

    //Put function in Async storage
    const getSwitchValue = async () => {
        let getAccessToken = await AsyncStorage.getItem("accessToken");
        let switchValue = await AsyncStorage.getItem("switchVal");
        if (switchValue == "true") {
            const resp = await getUsersCalendarList(getAccessToken);
            getFilteredGoogleCalendarList(resp);
        }
        setSwitchVal2(switchValue);
        setAccessToken(getAccessToken);
    };

    const signInOrOut = async (val) => {
        if (val) {
            const respCalendars = await signInWithGoogleAsync();
            if (respCalendars.error == true){
                setSwitchVal2("false");
            }
            else{
                getFilteredGoogleCalendarList(respCalendars);
            }
        }
        else {
            if (accessToken) {
                /* Log-Out */
                setCalendarList(null);
                await Google.logOutAsync({
                    accessToken, iosClientId: "128383090622-lgrk639fn4k6t99lhrldkh02441fcjgb.apps.googleusercontent.com",
                });
            }
        }
    }

    async function signInWithGoogleAsync() {
        try {
            const result = await Google.logInAsync({
                // androidClientId: YOUR_CLIENT_ID_HERE,
                iosClientId: "128383090622-lgrk639fn4k6t99lhrldkh02441fcjgb.apps.googleusercontent.com",
                scopes: ['profile', 'email', 'https://www.googleapis.com/auth/calendar.readonly'],
            });
            if (result.type === 'success') {
                AsyncStorage.setItem("accessToken", result.accessToken);
                setAccessToken(result.accessToken);
                return getUsersCalendarList(result.accessToken); //Here We are getting all the calendars in a json
            } else {
                return { cancelled: true };
            }
        } catch (e) {
            return { error: true };
        }
    }

    const getFilteredGoogleCalendarList = (respCalendars) => {
        var filteredList = respCalendars.items.map(element => {
            return { id: element.id, summary: element.summary, description: element.description };
        });
        setCalendarList(filteredList);
    };

    const getUsersCalendarList = async (accessToken) => {
        let calendarsList = await fetch('https://www.googleapis.com/calendar/v3/users/me/calendarList', {
            headers: { Authorization: `Bearer ${accessToken}` },
        });
        let resp = await calendarsList.json();
        return resp;
    }

    const press = (item) => { 
        var { checked } = isConnected;
        // These ensures that multiple checkboxes don't all get affected when one is clicked
        if (!checked.includes(item) || checked.length == 1) {
            checked = [];
            setIsConnected({ checked: [...checked, item] });
        } else {
            setIsConnected({ checked: checked.filter(a => a != item) });
        }
        AsyncStorage.setItem("calendarId", item);
    };

    /**
    * The method will slide the side menu from the right side of the screen
    * @param  {} =>{props.navigation.openDrawer(
    */
    const goToMenu = () => {
        props.navigation.openDrawer();
    };

    useEffect(() => {
        getSwitchValue();
    }, []);

    return (
        <View style={styles.container}>
            <View style={styles.menuButtonContainer}>
                <TouchableOpacity style={styles.menuButton} onPress={goToMenu}>
                    <Feather name="menu" style={styles.icon} />
                </TouchableOpacity>
            </View>
            <Text style={styles.mainLabel}>Settings</Text>
            <View style={styles.scrollContainer}>
                <ScrollView scrollEnabled={false}>
                    <View style={styles.container1}>
                        <Text style={styles.container1Text}>Notifications</Text>
                        <Text style={styles.container1SubText}>Current Status is {switchLabel1}</Text>
                        <View style={styles.toggle}>
                            <Switch
                                value={switchVal1}
                                onValueChange={(val) => setSwitchVal1(val)}>
                            </Switch>
                        </View>
                    </View>
                    <View style={styles.container2}>
                        <Text adjustsFontSizeToFit style={styles.container2Text}>Google Calendar Sync</Text>
                        <Text style={styles.container2SubText}>Current Status is {switchLabel2}</Text>
                        <View style={styles.toggle}>
                            <Switch
                                value={switchVal2 == "true" ? true : false}
                                onValueChange={(val) => {
                                    AsyncStorage.setItem("switchVal", JSON.stringify(val));
                                    setSwitchVal2(val.toString());
                                    signInOrOut(val);
                                }
                                }>
                            </Switch>
                        </View>
                    </View>
                </ScrollView>
            </View>
            <View style={styles.flatlist}>
                {(calendarList) &&
                    <FlatList
                        data={calendarList}
                        keyExtractor={(item) => item.id}
                        renderItem={({ item }) => (
                            <TouchableOpacity onPress={() => { press(item.id);
                             }}>
                                <ListItem
                                    title={item.summary}
                                    titleStyle={{ color: "white" }}
                                    subtitle={item.description !== undefined ? item.description : "Please provide a description inside google calendar"}
                                    rightTitleStyle={{ color: "green" }}
                                    subtitleStyle={{ color: "grey" }}
                                    containerStyle={{ backgroundColor: "#2A2E43" }}
                                    rightIcon={
                                        <CheckBox
                                            size={30}
                                            iconRight
                                            iconType='material'
                                            checkedIcon='check'
                                            uncheckedIcon='add'
                                            uncheckedColor="grey"
                                            checkedColor='#3ACCE1'
                                            onPress={() => {
                                                press(item.id);
                                            }}
                                            checked={isConnected.checked.includes(item.id)}
                                        />}
                                />
                            </TouchableOpacity>
                        )}
                    />
                }
            </View>
        </View>
    );
}

Settings.propTypes = {
    navigation: PropTypes.object,
    openDrawer: PropTypes.func,
};

const settingsStyle = {
    mainLabel: {
        color: "#FFFFFF",
        fontSize: 25,
        fontWeight: "bold",
        fontFamily: "encodeSansExpanded",
        marginTop: "5%",
        marginBottom: "5%"
    },
    container1: {
        width: "100%",
        height: "100%",
        backgroundColor: "#353A50",
        flexDirection: "column",
        justifyContent: "center"
    },
    container1Text: {
        color: "#FFF",
        marginHorizontal: "10%",
        fontSize: 18,
        fontFamily: "encodeSansExpanded"
    },
    container1SubText: {
        color: "#FFF",
        opacity: 0.3,
        marginHorizontal: "10%",
        fontFamily: "encodeSansExpanded"
    },
    container2: {
        width: "100%",
        height: "100%",
        backgroundColor: "#353A50",
        top: "20%",
        flexDirection: "column",
        justifyContent: "center"
    },
    container2Text: {
        color: "#FFF",
        marginHorizontal: "10%",
        fontSize: 18,
        fontFamily: "encodeSansExpanded"
    },
    container2SubText: {
        color: "#FFF",
        opacity: 0.3,
        marginHorizontal: "10%",
        fontFamily: "encodeSansExpanded"
    },
    scrollContainer: {
        height: "25%",
        width: "100%",
        backgroundColor: "#353A50",
        flexDirection: "column",
    },
    toggle: {
        position: "absolute",
        left: "80%",
        top: "30%"
    },
    flatlist: {
        height: "60%",
        width: "100%",
        marginLeft: 20,
        flexDirection: "column",
    }
}

export const styles = StyleSheet.create({...sideMenuStyle, ...settingsStyle});

export default Settings;
