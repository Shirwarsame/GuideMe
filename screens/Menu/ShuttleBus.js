import React from "react";
import { View, Text, StyleSheet} from "react-native";
import { Icon, Button } from "native-base";
import { Feather } from "@expo/vector-icons";



function ShuttleBus(props) {

    const goToMenu = () => {
        props.navigation.openDrawer();;
    };

    return (
        <View style={styles.container}>
            <View style={styles.menuButtonContainer}>
                <Button transparent style={styles.menuButton} onPress={goToMenu}>
                <Feather name="menu" style={styles.icon} />
                </Button>
            </View>
           

            <Text style={styles.mainLabel}>Shuttle Bus</Text>

        </View >
    );
}

export const styles = StyleSheet.create({
    container: {
        alignItems: "center",
        justifyContent: "space-between",
        height: "100%",
        width: "100%",
        backgroundColor: "#2A2E43"
    },
    mainLabel: {
        color: "#FFFFFF",
        position: "absolute",
        fontSize: 25,
        fontWeight: "bold",
        fontFamily: "encodeSansExpanded",
        top: "15%"
    },
    icon: {
            height: "100%",
            width: "100%",
            flexDirection: "row",
            left: "6%",
            color: "#FFFFFF",
            fontSize: 35

    },
    menuButton: {
        height: "100%",
        width: "100%",
        flexDirection: "row",
    },
    menuButtonContainer: {
        width: "100%",
        height: "6%",
        flexDirection: "column",
        justifyContent: "space-around",
        alignContent: "center",
        alignItems: "center",
        top: "7%"
    },

});

export default ShuttleBus;
