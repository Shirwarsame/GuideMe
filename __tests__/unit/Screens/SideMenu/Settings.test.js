import React from "react";
import {Settings} from "../../../../screens/Menu/Settings";
import renderer from "react-test-renderer";
import { TouchableOpacity, Switch } from "react-native";
import { shallow } from "enzyme";

describe("Settings component", () => {
    test("renders correctly", () => {
        const tree = renderer.create(<Settings />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test("click on components", () => {
        const navigation = { openDrawer: (param, defaultValue) => {
            return "NearbyInterest";
        }};

        const wrapper = shallow(<Settings navigation={navigation}/>);
        wrapper.find(TouchableOpacity).props().onPress();
        wrapper.find(Switch).at(0).simulate('valueChange')
    });
});