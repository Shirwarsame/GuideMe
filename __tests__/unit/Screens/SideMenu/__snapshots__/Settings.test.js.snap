// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2A2E43",
      "height": "100%",
      "justifyContent": "space-between",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "height": "6%",
        "top": "7%",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexDirection": "row",
          "height": "100%",
          "justifyContent": "center",
          "opacity": 1,
          "width": "20%",
        }
      }
    >
      <Text />
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontFamily": "encodeSansExpanded",
        "fontSize": 25,
        "fontWeight": "bold",
        "marginBottom": "5%",
        "marginTop": "5%",
      }
    }
  >
    Settings
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#353A50",
        "flexDirection": "column",
        "height": "25%",
        "width": "100%",
      }
    }
  >
    <RCTScrollView
      collapsable={false}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      scrollEnabled={false}
    >
      <View>
        <View
          style={
            Object {
              "backgroundColor": "#353A50",
              "flexDirection": "column",
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FFF",
                "fontFamily": "encodeSansExpanded",
                "fontSize": 18,
                "marginHorizontal": "10%",
              }
            }
          >
            Notifications
          </Text>
          <Text
            style={
              Object {
                "color": "#FFF",
                "fontFamily": "encodeSansExpanded",
                "marginHorizontal": "10%",
                "opacity": 0.3,
              }
            }
          >
            Current Status is 
            OFF
          </Text>
          <View
            style={
              Object {
                "left": "80%",
                "position": "absolute",
                "top": "30%",
              }
            }
          >
            <RCTSwitch
              accessibilityRole="button"
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "height": 31,
                  "width": 51,
                }
              }
              value={false}
            />
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#353A50",
              "flexDirection": "column",
              "height": "100%",
              "justifyContent": "center",
              "top": "20%",
              "width": "100%",
            }
          }
        >
          <Text
            adjustsFontSizeToFit={true}
            style={
              Object {
                "color": "#FFF",
                "fontFamily": "encodeSansExpanded",
                "fontSize": 18,
                "marginHorizontal": "10%",
              }
            }
          >
            Google Calendar Sync
          </Text>
          <Text
            style={
              Object {
                "color": "#FFF",
                "fontFamily": "encodeSansExpanded",
                "marginHorizontal": "10%",
                "opacity": 0.3,
              }
            }
          >
            Current Status is 
            OFF
          </Text>
          <View
            style={
              Object {
                "left": "80%",
                "position": "absolute",
                "top": "30%",
              }
            }
          >
            <RCTSwitch
              accessibilityRole="button"
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "height": 31,
                  "width": 51,
                }
              }
              value={false}
            />
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "flexDirection": "column",
        "height": "60%",
        "marginLeft": 20,
        "width": "100%",
      }
    }
  />
</View>
`;
